src/operator/gitea.ts

```diff
@@ -129,3 +165,349 @@ async function main(): Promise<void> {
+
+  // Check and execute setupGitea if dependencies changed
+  if (
+    !currentState.giteaPassword ||
+    !currentState.teamConfig ||
+    currentState.giteaPassword !== lastState.giteaPassword ||
+    currentState.teamConfig !== lastState.teamConfig
+  ) {
+    await setupGitea()
+  }
+
+  // Check and execute setGiteaOIDCConfig if dependencies changed
+  if (
+    !currentState.oidcClientId ||
+    !currentState.oidcClientSecret ||
+    !currentState.oidcEndpoint ||
+    currentState.oidcClientId !== lastState.oidcClientId ||
+    currentState.oidcClientSecret !== lastState.oidcClientSecret ||
+    currentState.oidcEndpoint !== lastState.oidcEndpoint
+  ) {
+    await setGiteaOIDCConfig()
+  }
+
+  // Check and execute setGiteaGroupMapping if dependencies changed
+  if (!currentState.teamNames || currentState.teamNames !== lastState.teamNames) {
+    await setGiteaGroupMapping('gitea', 'gitea-0')
+  }
+
+  // Update last known state
+  lastState = currentState
+}
+
+async function runSetupGitea() {
+  try {
+    await checkAndExecute()
+  } catch (error) {
+    console.debug('Error could not run setup gitea', error)
+    console.debug('Retrying in 30 seconds')
+    await new Promise((resolve) => setTimeout(resolve, 30000))
+    console.debug('Retrying to setup gitea')
+    await runSetupGitea()
+  }
+}
+
+// Setup Gitea Functions
+async function upsertTeam(
+  existingTeams: Team[] = [],
+  orgApi: OrganizationApi,
+  teamOption: CreateTeamOption,
+): Promise<void> {
+  const existingTeam = existingTeams.find((el) => el.name === teamOption.name)
+  if (existingTeam)
+    return doApiCall(
+      errors,
+      `Updating team "${teamOption.name}" in org "${orgName}"`,
+      () => orgApi.orgEditTeam(existingTeam.id!, teamOption),
+      422,
+    )
+  return doApiCall(
+    errors,
+    `Updating team "${teamOption.name}" in org "${orgName}"`,
+    () => orgApi.orgCreateTeam(orgName, teamOption),
+    422,
+  )
+}
+
+async function upsertRepo(
+  existingRepos: Repository[] = [],
+  orgApi: OrganizationApi,
+  repoApi: RepositoryApi,
+  repoOption: CreateRepoOption | EditRepoOption,
+  teamName?: string,
+): Promise<void> {
+  const existingRepo = existingRepos.find((el) => el.name === repoOption.name)
+  if (!existingRepo) {
+    // org repo create
+    await doApiCall(
+      errors,
+      `Creating repo "${repoOption.name}" in org "${orgName}"`,
+      () => orgApi.createOrgRepo(orgName, repoOption as CreateRepoOption),
+      422,
+    )
+  } else {
+    // repo update
+    await doApiCall(
+      errors,
+      `Updating repo "${repoOption.name}" in org "${orgName}"`,
+      () => repoApi.repoEdit(orgName, repoOption.name!, repoOption as EditRepoOption),
+      422,
+    )
+  }
+  // new team repo, add team
+  if (teamName)
+    await doApiCall(
+      errors,
+      `Adding repo "${repoOption.name}" to team "${teamName}"`,
+      () => repoApi.repoAddTeam(orgName, repoOption.name!, teamName),
+      422,
+    )
+  return undefined
```

Comment:
return undefined is the default right so it can be removed?

Reply:
Removed!